### ISSUE TITLE
Clang 15.0.2-1 has crushed when compiling source code.

### ISSUE THREAD

**Issue by v-for-vandal** [llvm/llvm-project#60359]
As requested in console output, I am attaching stack dump and all requested files.

https://pastebin.com/zuhfefxC

```
> clang --version
Ubuntu clang version 15.0.2-1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by v-for-vandal** [comment]
[world-b50451.sh.txt](https://github.com/llvm/llvm-project/files/10528355/world-b50451.sh.txt)
[world-b50451.cpp.txt](https://github.com/llvm/llvm-project/files/10528357/world-b50451.cpp.txt)

Appologies, I had to add '.txt' suffix to filenames.

**Comment by EugeneZelenko** [comment]
Could you please try `main` branch? https://godbolt.org should be helpful.

**Comment by DimitryAndric** [comment]
This has been fixed with 133b6d7db90d9b52b01e8e09e1aa8fb8d2da0f9d ("[clang][C++20] Fix clang/clangd assert/crash after compilation errors") by @dmpolukhin, to solve #53628. It was apparently never merged back to the 15.x release branch.
